### DISCUSSION TITLE
Help with Send Request 'Then'

### DISCUSSION THREAD

**Discussion by AkikoOrenji** [synfron/ReshaperForBurp#11] [General]
Great plugin by the way. Just learning it at the moment and its so flexable. 
I notice when i paste the request i want to send into the box marked as 'Request' it gets sent as a single line. I can't see any way to add line breaks to create a valid request.  The box wont accept using the Enter key and the text \r\n doesn't work either. Am i missing something ?

**Comment by AkikoOrenji** [comment]
I decided to store the request as a variable which allows for the new lines. The problem i now have is that the plugin seems to add these as newlines only \n which are dropped (should be \r\n). I tried a few different ways of adding \r\n e.g. using set variable with Regex replace for \n to \n\r  but this didn't work either. Any ideas ?  (is this beacuse i'm running burp on Windows ? )

**Comment by ddwightx** [comment]
Glad you like the extension!

Right, you do need to use a variable to enter text that has needs new lines, but I'll have to do some testing to see why carriage returns are being stripped. They aren't purposely being stripped. If you paste the text in a text editor and save as a file and reopen, can you tell if the carriage return is retained?

Also interesting that the Set Variable replace workaround didn't help. Another possible workaround is to replace in JavaScript using Then Run Script, or use Then Run Process to get the text from a file and capture it to a variable. Hopefully one of those helps for now. I will look further into this when I get a chance later.

**Comment by AkikoOrenji** [comment]
Stragely if i copy the text from burp to notepad++ it shows as \r\n yet when copied back in again it ends up as \n when being sent. 
Thanks for the other pointers.

**Comment by ddwightx** [comment]
Looks like it's due to the text area stripping \r characters when pulling text from it. Going to make some changes to how text is pulled from it.

The best workaround for now is to add a Run Process Then directly before the Send Request Then and retrieve the request from a file.
Set the following values (do not leave the <> characters):

Command - `cmd /k "type C:\path\to\file\containing\text.txt && exit"`
Wait for Completion - `Checked`
Capture Output - `Checked`
Capture Variable Source  -`<Global or Event>`
Capture Variable Name - `<Name of Variable>`

Then in Send Request:

Request - `{{<global or event>:<Name of Variable>}}`

**Reply by ddwightx** [reply]
This will be fixed in the next release. Here's a test build https://github.com/synfron/ReshaperForBurp/releases/tag/v1.4.0-next

**Reply by AkikoOrenji** [reply]
Thanks so much.
